src/peer-connection.ts

```diff
@@ -285,6 +285,45 @@ class PeerConnection extends EventEmitter<PeerConnectionEventHandlers> {
   get iceGatheringState(): RTCIceGathererState {
     return this.pc.iceGatheringState;
   }
+
+  /**
+   * Returns the type of a connection that has been established.
+   *
+   * @returns The connection type which would be {Promise<'UDP' | 'TCP' | 'TURN-TLS' | 'TURN-TCP' | 'TURN-UDP' | 'unknown'>}.
+   */
+  async getCurrentConnectionType(): Promise<string> {
+    // make sure this method only can be called when the ice connection is established;
+    const isIceConnected =
+      this.pc.iceConnectionState === 'connected' || this.pc.iceConnectionState === 'completed';
```

Comment:
Our PeerConnection class has a `getConnectionState()` method which we could use here and then you would only need to check against `ConnectionState.Connected` here. However, the `getConnectionState()` method returns the overall state (so it only says "connected" if both ICE and DTLS are connected) while here I think we only care about ICE state, so I don't mind that much which way we do it here, just writing the comment to make you aware that you have the option to use `PeerConnection.getConnectionState()` if you want to

Reply:
Thank for the reminder, yeah I see this API in `ConnectionStateHandler`, but as you said this `ConnectionState.Connected` states also represent the DTLS connection state which not actually meet our expectation, so we can just keep the code currently.